sql/table.go

```diff
@@ -365,8 +394,35 @@ func convertDatum(col ColumnDescriptor, val parser.Datum) (interface{}, error) {
 			return float64(v), nil
 		}
 	// case ColumnType_DECIMAL:
-	// case ColumnType_DATE:
-	// case ColumnType_TIMESTAMP:
+	case ColumnType_DATE:
+		switch val := val.(type) {
+		case parser.DString:
+			d, err := parser.ParseDate(val)
+			if err != nil {
+				return time.Time{}, err
+			}
+			return d.Time, nil
+		case parser.DDate:
```

Comment:
Should you also allow `DTimestamp` here and `DDate` below in the `TIMESTAMP` case?